torch/csrc/jit/codegen/cuda/utils.h

```diff
@@ -74,6 +76,118 @@ class PolymorphicBase {
   }
 };
 
+//! Container class DisjointSet models equivalence relationships
```

Comment:
Can we move this to a dedicated header? I think we should keep utils.h for very small things.

Also, this used to be a private helper. If we promote it to reusable it would make sense to add some unit tests, what do you think?

Reply:
It's going to be used in a new class I'm currently working on. Since it's going to need to be in a header file, so I decided to move it to the utility header. 

I'll move it to an individual header and add some tests. Thanks for your feedback!